File: ALGORITHMS_TRAINE/SELECTION_SORT/selectionSort.c
#include <stdio.h>
#include <stdlib.h>

void sortItem(int *, int *);

int main()
{
	int numb[5];
	int size = (int)(sizeof(numb) / sizeof(numb[0]));
   

	for (int i = 0; i < 5; i++)
	{
		printf("insert the %d elements of the array \n",i);
		scanf("%d", &numb[i]);
	}

	for (int i = 0; i < size; i++)
	{
		int * element = &numb[i];
		
		for (int j = i+1; j < size; j++)
		{
			if (numb[j] < numb[i])
			{
				element = &numb[j];
				
			}
		}
		
		sortItem(element, &numb[i]);
		
	}

	for (int i = 0; i < 5; i++)
	{
		printf("sorted %d \n", numb[i]);
		
	}

	return 0;
}

void sortItem(int *less, int *gret)
{
	
	int aux = *less;
	*less = *gret;
	*gret = aux;
}

File: ALGORITHMS_TRAINE/SELECTION_SORT/selectionSort.js
let array = [ 4,2,7,8,45,3,95]
function selectionSort(arr){

for(let i =0; i < arr.length -1;i ++){
	for(let j =i+1; j < arr.length-1;j ++){
		if(arr[j] < arr[i]){
		let aux = arr[i]
		arr[i]=arr[j]
		arr[j]=aux
			}
		}
	}
	return arr
}

selectionSort(array)
console.log(array)

